### ISSUE TITLE
Set resource collection based on user

### ISSUE THREAD

**Issue by 7200rpm** [benwinding/react-admin-firebase#137]
Hi,

I'd like to be able to set my resource collection based on the logged in user (e.g. users/Jane), but can't seem to figure out a way to do this easily. I see you can set the `rootref` in the options, but this would only be static. I need to set this dynamically based on who is logging in..

Seems like this would be a common use case where someone has different documents for each user. Am I missing something? 

Thanks!

**Comment by splex7** [comment]
In other words, do you want to create and control the firestore collection of users accessed via Authentication?

**Comment by splex7** [comment]
In other words, do you want to create and control the firestore collection of users accessed via Authentication?

**Comment by 7200rpm** [comment]
> In other words, do you want to create and control the firestore collection of users accessed via Authentication?

Yes, and access only the resources that are owned by the logged in user.

**Comment by seranotannason** [comment]
Hey @7200rpm, I agree that this would be a very useful feature. I think this issue is a more general version of our problem: https://github.com/benwinding/react-admin-firebase/issues/7. Essentially, accessing a resource sub-collection under the user (e.g. users/Jane/some-sub-collection) is not possible under react-admin-firebase yet.

**Comment by splex7** [comment]
@7200rpm. 
There is no user collection base on Firebase Authentication.In my case, when a user is created through Firebase Authentication, the function is triggered to create a new document in the users collection of firestore with the corresponding email. 
this may help you
https://bigcodenerd.org/create-user-profile-firestore-authentication/

**Comment by dalwadani** [comment]
This is how I did it:

```
let uid;
const authProvider = FirebaseAuthProvider(config, { logging: true });
const dataProvider = FirebaseDataProvider(config, {
  logging: true,
  rootRef: () => {
    authProvider.getAuthUser().then((data) => {
      uid = data.uid;
    });

    return `accounts/${uid}`;
  },
```

Please note that this code is not optimized and can be improved.
This is just a quick and dirty way to do it.
